doc/howto/container_gpu_passthrough_with_docker.md

```diff
@@ -0,0 +1,147 @@
+Consider the following scenario:
+
+Your host is an NVIDIA single board computer that has a Tegra SoC with an iGPU, and you have the Tegra SDK installed on the host. You want to create a LXD container and run an application inside the container using the iGPU as a compute backend. You want to run this application inside a Docker container (or another OCI-compliant runtime).
+To achieve this, complete the following steps:
+
+1. Initialize a LXD container with {config:option}`instance-security:security.nesting` enabled (needed for running a Docker container inside a LXD container):
+
+        lxc init ubuntu:24.04 t1 --config security.nesting=true
+
+1. Add the GPU device to your container:
+
+    - If you want to do an iGPU pass-through:
+
+          lxc config device add t1 igpu0 gpu gputype=physical id=nvidia.com/igpu=0
+
+    - If you want to do a dGPU pass-through:
+
+          lxc config device add t1 gpu0 gpu gputype=physical id=nvidia.com/gpu=0
+
+After adding the device, let's try to run a basic [MNIST](https://en.wikipedia.org/wiki/MNIST_database) inference job inside our LXD container.
+
+1. Create a `cloud-init` script that installs the Docker runtime, the [NVIDIA Container Toolkit](https://github.com/NVIDIA/nvidia-container-toolkit), and a script to run a test [TensorRT](https://github.com/NVIDIA/TensorRT) workload:
+
+        #cloud-config
+        package_update: true
+        packages:
+          - docker.io
+        write_files:
+          - path: /root/run_tensorrt.sh
+            permissions: "0755"
+            owner: root:root
+            content: |
+            #!/bin/bash
+            echo "OS release,Kernel version"
+            (. /etc/os-release; echo "${PRETTY_NAME}"; uname -r) | paste -s -d,
+            echo
+            nvidia-smi -q
+            echo
+            exec bash -o pipefail -c "
+            cd /workspace/tensorrt/samples
+            make -j4
+            cd /workspace/tensorrt/bin
+            ./sample_onnx_mnist
+            retstatus=\${PIPESTATUS[0]}
+            echo \"Test exited with status code: \${retstatus}\" >&2
+            exit \${retstatus}
+            "
+        runcmd:
+          - systemctl start docker
+          - systemctl enable docker
+          - usermod -aG docker root
+          - curl -fsSL https://nvidia.github.io/libnvidia-container/gpgkey | sudo gpg
+            --dearmor -o /usr/share/keyrings/nvidia-container-toolkit-keyring.gpg
+          - curl -s -L
+            https://nvidia.github.io/libnvidia-container/stable/deb/nvidia-container-toolkit.list
+            | sed 's#deb https://#deb
+            [signed-by=/usr/share/keyrings/nvidia-container-toolkit-keyring.gpg]
+            https://#g' | tee /etc/apt/sources.list.d/nvidia-container-toolkit.list
+          - sed -i -e '/experimental/ s/^#//g'
+            /etc/apt/sources.list.d/nvidia-container-toolkit.list
+          - apt-get update
```

Comment:
Would it make sense to do this first, and then install docker as part of this block rather than via the `packages` statement, to keep it all together?

Reply:
fixed